dashboard/app/controllers/registrations_controller.rb

```diff
@@ -59,6 +59,7 @@ def upgrade
     successfully_updated = can_update && current_user.update(update_params(params_to_pass))
     has_email = current_user.parent_email.blank? && current_user.hashed_email.present?
     success_message_kind = has_email ? :personal_login_created_email : :personal_login_created_username
+    user.reload unless successfully_updated # if update fails, roll back user model so error page renders correctly
```

Comment:
That doesn't appear to be the case, the errors show up correctly with this change:

<img width="612" alt="screen shot 2017-07-13 at 1 17 23 pm" src="https://user-images.githubusercontent.com/70630/28185851-c354f3b2-67cd-11e7-9775-9cac5d2749e8.png">

What the user typed in gets cleared out, which is not ideal, but in this context (just the upgrade call) it's not terrible because most of that information is either secret confirmation or password-related, so doesn't auto-fill anyways.